### ISSUE TITLE
SQL Custom Field: default value is broken because it can't be validated

### ISSUE THREAD

**Issue by caru80** [joomla/joomla-cms#23568]
### Steps to reproduce the issue

1. Add a custom field of type sql in com_fields with a query that delivers an actual result.
2. Enter a default value that you know it exists in the result list.
3. Hit save.

### Expected result

The custom field get's stored.

### Actual result

Error Message „COM_FIELDS_FIELD_INVALID_DEFAULT_VALUE” caused by method checkDefaultValue in FieldsModelField of com_fields.

### System information (as much as possible)

Joomla! 3.9.2

### Additional comments

FieldsModelField::checkDefaultValue runs into line 275 (as it is supposed to do)...: 
```php
$result = $rule->test(simplexml_import_dom($node->firstChild), $data['default_value']);
```
... and \Joomla\CMS\Form\Rule\OptionsRule->test() expects that \SimpleXMLElement $element has some options where the default value can (or not) be found in, in line 89:

```php
return in_array((string) $value, $options);
```
But \SimpleXMLElement $element has no options (the result of the SQL Query). Thus it will allways fail.

**Comment by ghost** [comment]
@HLeithner  can you please comment?

**Comment by ghost** [comment]
@HLeithner gently Reminder.

**Comment by HLeithner** [comment]
if @caru80 would be so kind and write a PR we could test it and merge it

**Comment by ReLater** [comment]
The reason is the same like in field itemlist. It extends `FieldsListPlugin`.

https://github.com/joomla/joomla-cms/blob/staging/plugins/fields/sql/sql.php#L19

Thus the validation runs through `OptionsRule.php` while saving without options. ==> always false

See https://github.com/joomla/joomla-cms/pull/24643 . I've opened it just because the original issue https://github.com/joomla/joomla-cms/issues/15173 is > 2 years old or so without any clarifying statements.

But before we start to write several prs like that and maybe open new security issues it would be nice if someone of the "com_fields experts" would issue a statement if that's the correct solution or if we need new BlablaRules.php or whatever.

**Comment by ghost** [comment]
> But before we start to write several prs like that and maybe open new security issues it would be nice if someone of the "com_fields experts" would issue a statement if that's the correct solution or if we need new BlablaRules.php or whatever.

@laoneo ?

**Comment by ghost** [comment]
Closed as having Pull Request #24755
